File: api/src/main/java/database/summoner/SummonerDAOContract.java
package database.summoner;

import model.SummonerDetails;

/**
 * @author Josiah Kendall
 */
public interface SummonerDAOContract {
    public static final String SUMMONER_TABLE = "Summoner";
    public static final String ID = "Id";
    public static final String ACCOUNT_ID = "AccountId";
    public static final String SUMMONER_NAME = "SummonerName";
    public static final String SUMMONER_LEVEL = "SummonerLevel";
    public static final String PROFILE_ICON_ID = "ProfileIconId";
    public static final String REVISION_DATE = "RevisionDate";

    long saveSummoner(SummonerDetails summoner);
    void deleteSummoner(long summonerId);
    SummonerDetails getSummoner(long summonerId);
}

File: api/src/main/java/database/summoner/SummonerDAOContract.kt
package database.summoner

import model.response_beans.SummonerDetails

/**
 * @author Josiah Kendall
 */
interface SummonerDAOContract {

    fun saveSummoner(summoner: SummonerDetails): Long
    fun deleteSummoner(summonerId : String)
    fun getSummoner(summonerId : String): SummonerDetails?

    companion object {
        val SUMMONER_TABLE = "Summoner"
        val ID = "Id"
        val ACCOUNT_ID = "AccountId"
        val SUMMONER_NAME = "SummonerName"
        val SUMMONER_LEVEL = "SummonerLevel"
        val PROFILE_ICON_ID = "ProfileIconId"
        val REVISION_DATE = "RevisionDate"
    }
}
